integration-tests/cypress/cypress.spec.js

```diff
@@ -231,22 +230,27 @@ moduleTypes.forEach(({
         testOutput += chunk.toString()
       })
 
-      childProcess.on('exit', () => {
+      await Promise.all([
+        once(childProcess.stdout, 'end'),
+        once(childProcess.stderr, 'end'),
+        once(childProcess, 'exit'),
+        eventsPromise
+      ])
+
+      assert.isFalse(hasReceivedEvents)
```

Comment:
Nit: please only use assert methods that are available by Node.js assert as well. That way a later migration becomes easier.

```suggestion
      assert.strictEqual(hasReceivedEvents, false)
```